DM-5341: Changelog for lwm2m endpoint authorization (#4786)
SoftwareAG/c8y-docs: content/change-logs/device-management/lwm2m-agent-enhanced-access-control-and-permissions.md

---
date: ""

> title: Enhanced access control and permissions for LWM2M agents

Reviewer: ```suggestion
title: Enhanced access control and permissions for LWM2M service
```

product_area: Device management & connectivity
change_type:
  - value: change-QHu1GdukP
    label: Feature
component:
  - value: component-1KLUzmqfe
    label: LWM2M
build_artifact:
  - value: tc-ggH2M4hf3
    label: lwm2m-agent
ticket: DM-5341
version: 1022.3.0
---

> To improve security for the LWM2M agent, a stricter access control is enforced. To work with device-specific configurations, including **Device settings**, **Connectivity**, and **Bootstrap**, etc. and object mappings, users must now be granted explicit LWM2M agent permissions. Administrators can assign the necessary permissions to users in **Administration** > **Accounts** > **Users**, **Roles**.

Reviewer: ```suggestion
To improve security for the LWM2M service, a stricter access control is enforced. To work with device-specific configurations from the Device management application, including **Device settings**, **Connectivity**, and **Bootstrap**, etc., **Device protocols**, **Device registration**, users must now be granted explicit LWM2M Admin and Read permissions. Administrators can assign the necessary permissions to users in **Administration** > **Accounts** > **Users**, **Roles**.
```

> To improve security for the LWM2M service, a stricter access control is enforced. To work with device-specific configurations from the Device management application, including **Device settings**, **Connectivity**, and **Bootstrap**, etc., **Device protocols**, **Device registration**, users must now be granted explicit LWM2M Admin and Read permissions. Administrators can assign the necessary permissions to users in **Administration** > **Accounts** > **Users**, **Roles**.

Reviewer 2: ```suggestion
To improve security for the LWM2M service, stricter access control has been enforced. To work with device-specific configurations from the Device management application, including **Device settings**, **Connectivity**, and **Bootstrap**, **Device protocols**, **Device registration**, users must now be granted explicit LWM2M ADMIN and READ permissions. Administrators can assign the necessary permissions to users in **Administration** > **Accounts** > **Users** > **Roles**.
```

Reviewer 2: Not sure about this listing. I removed "etc" in the middle of the list, but still it sounds messy.

Author: @BeateRixen I updated the text a little bit. What I wanted to capture here is... we have LWM2M configuration tab under each device which contains the following groups as sub-tabs: Device settings. Connectivity, Bootstrap and so on. To view or update those configurations users will need the new permissions.
Moreover, also the 'Device protocols' which is on tenant-level requires the new lwm2m permissions.
any suggestions on how to rephrase this?

Reviewer 2: @sourajitsaha17 How about my changes/proposal?

Author: @BeateRixen Thank you :) I like it very much. Can I merge it?

Reviewer 2: @sourajitsaha17 I finally merged it.

> To improve security for the LWM2M service, stricter access control has been enforced. To work with device-specific LWM2M configurations from the Device management application, such as **Device settings**, **Connectivity**, and **Bootstrap**, **Device registration**, and tenant-level **Device protocols**, users must now be granted explicit LWM2M ADMIN and READ permissions. Administrators can assign the necessary permissions to users in **Administration** > **Accounts** > **Users** > **Roles**.

Reviewer 2: ```suggestion
To improve security for the LWM2M service, stricter access control has been enforced. To work with device-specific LWM2M configurations from the Device Management application, users must now be granted explicit LWM2M ADMIN and READ permissions. This affects **Device settings**, **Connectivity**, **Bootstrap**, and more under the **LWM2M Configuration** tab, the device registration, and **Device protocols** at the tenant level. Administrators can assign the necessary permissions to users in **Administration** > **Accounts** > **Users** > **Roles**.
```
